src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -241,25 +241,28 @@ private void importRoslynReport(String reportPath) {
     }
 
     JsonParser parser = new JsonParser();
-    for (JsonElement issueElement : parser.parse(contents).getAsJsonObject().get("issues").getAsJsonArray()) {
-      JsonObject issue = issueElement.getAsJsonObject();
-
-      String ruleId = issue.get("ruleId").getAsString();
-      if (!activeRuleIds.contains(ruleId)) {
-        continue;
-      }
+    JsonElement issues = parser.parse(contents).getAsJsonObject().get("issues");
+    if (issues != null) {
+      for (JsonElement issueElement : issues.getAsJsonArray()) {
+        JsonObject issue = issueElement.getAsJsonObject();
+
+        String ruleId = issue.get("ruleId").getAsString();
+        if (!activeRuleIds.contains(ruleId)) {
+          continue;
+        }
 
-      String message = issue.get(issue.has("shortMessage") ? "shortMessage" : "fullMessage").getAsString();
-      for (JsonElement locationElement : issue.get("locations").getAsJsonArray()) {
-        JsonObject location = locationElement.getAsJsonObject();
-        if (location.has("analysisTarget")) {
-          for (JsonElement analysisTargetElement : location.get("analysisTarget").getAsJsonArray()) {
-            JsonObject analysisTarget = analysisTargetElement.getAsJsonObject();
-            String uri = analysisTarget.get("uri").getAsString();
-            JsonObject region = analysisTarget.get("region").getAsJsonObject();
-            int startLine = region.get("startLine").getAsInt();
-
-            handleRoslynIssue(ruleId, uri, startLine, message);
+        String message = issue.get(issue.has("shortMessage") ? "shortMessage" : "fullMessage").getAsString();
+        for (JsonElement locationElement : issue.get("locations").getAsJsonArray()) {
+          JsonObject location = locationElement.getAsJsonObject();
+          if (location.has("analysisTarget")) {
+            for (JsonElement analysisTargetElement : location.get("analysisTarget").getAsJsonArray()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS134)